codegenerator/cli/npm/envio/src/FetchState.res

```diff
@@ -953,45 +929,43 @@ let getNextQuery = (
   }
 }
 
-type itemWithPopFn = {item: Internal.item, popItemOffQueue: unit => unit}
-
-/**
-Represents a fetchState partitions head of the  fetchedEventQueue as either
-an existing item, or no item with latest fetched block data
-*/
-type queueItem =
-  | Item(itemWithPopFn)
-  | NoItem({latestFetchedBlock: blockNumberAndTimestamp})
-
-/**
-Simple constructor for no item from partition
-*/
-let makeNoItem = ({latestFetchedBlock}: partition) => NoItem({
-  latestFetchedBlock: latestFetchedBlock,
-})
+let getTimestampAt = (fetchState: t, ~index) => {
+  switch fetchState.buffer->Belt.Array.get(index) {
+  | Some(Event({timestamp})) => timestamp
+  | Some(Block(_)) =>
+    Js.Exn.raiseError("Block handlers are not supported for ordered multichain mode.")
+  | None => (fetchState->bufferBlock).blockTimestamp
+  }
+}
 
-/**
-Gets the earliest queueItem from thgetNodeEarliestEventWithUpdatedQueue.
+let hasReadyItem = ({buffer} as fetchState: t) => {
+  switch buffer->Belt.Array.get(0) {
+  | Some(item) => item->Internal.getItemBlockNumber <= fetchState->bufferBlockNumber
+  | None => false
+  }
+}
 
-Finds the earliest queue item across all partitions and then returns that
-queue item with an update fetch state.
-*/
-let getEarliestEvent = (fetchState: t) => {
-  let {queue} = fetchState
-  switch fetchState.queue->Utils.Array.last {
-  | Some(item) =>
-    if item->Internal.getItemBlockNumber <= fetchState->bufferBlockNumber {
-      Item({item, popItemOffQueue: () => queue->Js.Array2.pop->ignore})
-    } else {
-      NoItem({
-        latestFetchedBlock: fetchState->bufferBlock,
-      })
+let getReadyItemsCount = (fetchState: t, ~targetSize: int, ~fromItem) => {
```

Comment:
Is this only used in the test? Maybe should be a comment then or added to the naming

Reply:
Nvmnd I see where its used